### ISSUE TITLE
schema id ignored { type: 'string' } getting this warning

### ISSUE THREAD

**Issue by goel-sunny** [ajv-validator/ajv#987]
Hi,

I am using this module for the JSON validation , my JSON got validated and i got the desired result but i am confused why  i am getting this error. 

does id is necessary in the schema . because i am not specifying the id in the schema .

if is necessary then why ?

**Comment by epoberezkin** [comment]
You may want to use the option `{schemaId: 'auto'}` (see the docs: https://github.com/epoberezkin/ajv#using-version-6)

**Comment by cyberwombat** [comment]
I'm still unclear. I get the same warning. I wish to use "id" as part of my data to validate and it has nothing to do with the schema ID. Docs indicate the default is "$id" so why the warning:
```
$id: '...',
  title: 'Log Entry',
  type: 'object',
  dynamicDefaults: { id: 'uuid' },
  properties: {
    id: { type: 'string', format: 'uuid' } // this id is a whole different thing
   ...
```
The `schemaId` options in docs seem to refer to the topmost `$id` property. I am not clear why I thinks that an inner `id` is related.

**Comment by epoberezkin** [comment]
@cyberwombat it may be a separate issue or it may be that you are using the old version (or it may be caused by dynamicDefaults). Please investigate and submit the new issue following the issue template. If the issue is caused by dynamicDefaults it should be submitted to ajv-keywords package rather than here.
